gwpy/types/tests/test_series.py

```diff
@@ -167,6 +167,32 @@ def test_getitem(self, array):
                 name=array.name, epoch=array.epoch, unit=array.unit),
             )
 
+    def test_getitem_index(self, array):
+        """Test that __getitem__ also applies to an xindex.
+
+        When subsetting a Series with an iterable of integer indices,
+        make sure that the xindex, if it exists, is also subsetted. Tests
+        regression against https://github.com/gwpy/gwpy/issues/1680.
+        """
+        xindex = array.xindex  # create xindex
```

Comment:
This value doesn't need to be assigned:

```suggestion
        array.xindex  # create xindex
```